packages/user-interface/src/contexts/NotificationContext.tsx

```diff
@@ -0,0 +1,65 @@
+import { AlertProps } from "@gemeente-denhaag/alert";
+import { createContext, ReactNode, useEffect, useState } from "react";
+import Notification from "../components/Notification";
+import { useLocation, useSearchParams } from "react-router-dom";
+
+interface NotificationContextType {
+  getNotifications: () => ReactNode[];
+  pushNotification: (key: string, alert: AlertProps) => void;
+  checkUrlParam: (key: string, value: string) => boolean;
+}
+
+const NotificationContext = createContext<NotificationContextType>(
+  {} as NotificationContextType,
+);
+
+interface NotificationProviderProps {
+  children: ReactNode;
+}
+
+const INITIAL_MAP = new Map<string, ReactNode>();
+
+export const NotificationProvider = ({
+  children,
+}: NotificationProviderProps) => {
+  const [notifications, setNotifications] =
+    useState<Map<string, ReactNode>>(INITIAL_MAP);
+  const location = useLocation();
+  const [searchParams] = useSearchParams();
+
+  useEffect(() => {
+    setNotifications(INITIAL_MAP);
+  }, [location]);
+
+  const pushNotification = (key: string, alert: AlertProps) => {
+    setNotifications((prevNotifications) => {
+      return new Map<string, ReactNode>([...prevNotifications]).set(
+        key,
+        <Notification key={key} {...alert} />,
+      );
+    });
+  };
+
+  const getNotifications = () => {
+    return Array.from(notifications.values());
+  };
+
+  const checkUrlParam = (key: string, value: string) => {
```

Comment:
Wat doet deze functie?

Reply:
Heb m verwijderd, had die toegevoegd om de notify url param makkelijk te kunnen checken, maar zoals gisteren besproken proberen we het nu via de state te gaan doen.